src/components/graph/menus/create-node-menu.js

```diff
@@ -268,26 +307,33 @@ const CreateNodeMenu = ({
             onClose={handleClose}
         >
             {Object.values(NODE_MENU_ITEMS).map((item) => {
+                const hasSubMenuItem = item.subMenuItems !== undefined;
                 return (
                     (activeNode?.type !== 'ROOT' || item.onRoot) && (
-                        <MenuItem
-                            className={classes.menuItem}
-                            onClick={item.action}
-                            key={item.id}
-                            disabled={item.disabled}
-                        >
-                            <ListItemText
-                                key={item.id}
-                                className={classes.listItemText}
-                                primary={
-                                    <Typography noWrap>
-                                        {intl.formatMessage({
-                                            id: item.id,
-                                        })}
-                                    </Typography>
-                                }
-                            />
-                        </MenuItem>
+                        <div key={item.id}>
+                            {!hasSubMenuItem && <SubMenuItem item={item} />}
+
+                            {hasSubMenuItem && (
+                                <NestedMenuItem
+                                    label={intl.formatMessage({
+                                        id: item.id,
+                                    })}
+                                    parentMenuOpen={true}
+                                    disabled={item.disabled}
+                                >
+                                    {Object.values(item.subMenuItems).map(
+                                        (item) => {
```

Comment:
use another name here to avoid confusion with the existing `item` variable.
Something like `(subItem)` should be ok